utils/error_utils.py

```diff
@@ -0,0 +1,17 @@
+import os
+import re
+
+
+def reformat_error(e) -> str:
+    err = str(e)
+    if "session_id is no longer usable" in err:
+        err = "Session expired. Please refresh the page."
+    else:
+        # Regex pattern to match on "Insufficient privileges:" or "Permission denied:"
+        # and replace "User" with the service principal id
+        pattern = r"(.*(?:Insufficient privileges:|PERMISSION_DENIED:)\s)(User)(.*)"
+        replacement = (
+            r"\1Service principal id " + os.getenv("DATABRICKS_CLIENT_ID") + r"\3"
```

Comment:
is this to make sure the user understands it's the SP that doesn't have access, not them?

Reply:
yeah, i feel like the default message of "User" is pretty misleading unless they know how lakehouse apps work